homeassistant/config.py

```diff
@@ -669,6 +680,128 @@ def async_process_component_config(hass, config, domain):
     return None
 
 
+def check_ha_config_file(config_dir):
+    """Check if Home Assistant configuration file is valid."""
+    CheckConfigError = namedtuple(  # pylint: disable=invalid-name
+        'CheckConfigError', "message domain config")
+
+    class ConfigResult(OrderedDict):
```

Comment:
We should not define this inside function

Reply:
With attr

Reply:
Will also move it to check_config.py.

@pvizeli not sure I'm following you?

Reply:
I think he means using the new [attr library](http://www.attrs.org/en/stable/) which has recently been added as a global requirement (https://github.com/home-assistant/architecture/issues/4)

Reply:
I think he means using the new [attrs library](http://www.attrs.org/en/stable/) which had recently been added as a global requirement (https://github.com/home-assistant/architecture/issues/4)

Reply:
Ok, thanks. And @attr.s will call super() for me?